realtime/utils.js

```diff
@@ -401,14 +401,15 @@ function attachAspectSubject(sample) {
      * memory issues.
      */
     const getAspectPromise = sample.aspect ? Promise.resolve(sample.aspect) :
-    sequelize.query('SELECT * FROM "Aspects" WHERE id = :aspectId and ' +
-      '"isDeleted" = 0', { replacements: { aspectId: sample.aspectId },
+    sequelize.query('SELECT * FROM "Aspects" WHERE name iLike :name and ' +
+      '"isDeleted" = 0 LIMIT 1', { replacements: { name: aspName },
       type: sequelize.QueryTypes.SELECT, });
 
     const getSubjectPromise = sample.subject ? Promise.resolve(sample.subject) :
     sequelize.query('SELECT id, "absolutePath", name, tags FROM "Subjects" ' +
-      'WHERE id = :subjectId and "isDeleted" = 0', { replacements:
-        { subjectId: sample.subjectId }, type: sequelize.QueryTypes.SELECT, });
+      'WHERE "absolutePath" iLike :subName and "isDeleted" = 0 LIMIT 1',
+      { replacements: { subName: subName },
+      type: sequelize.QueryTypes.SELECT, });
```

Comment:
Or, should we throw an error if subjectId and aspectId is not there, like L387?

Reply:
Log the error and move on, like we have for samples without name.

Reply:
Yeah I think we should log them - that way we at least have some way of knowing that there's a data problem